include/nbl/builtin/hlsl/workgroup/shared_scan.hlsl

```diff
@@ -167,9 +152,9 @@ struct WorkgroupScanTail
             firstLevelScan = glsl::subgroupShuffleUp(firstLevelScan, 1u);
```

Comment:
the barrier above should actually go on what is now line 146 at the end of the `lastInvocation>=glsl::gl_SubgroupSize()` clause, obviously under an extra `if (isExclusive)`

because if your workgroup is narrower than a subgroup, then there's no need to workgroup barrier because no subgroup has stored to `sharedAccessor` and you won't be `get`-ing anything from it later on

Reply:
hmm actually between this barrier and the previous barrier there's only a `get` so this barrier is not needed at all! remove it!